src/main/java/seedu/address/logic/commands/AddPatientCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DOB;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Patient;
+
+/**
+ * Adds a person to the address book.
+ */
+public class AddPatientCommand extends Command {
+
+    public static final String COMMAND_WORD = "addpatient";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a patient to the address book. "
+            + "Parameters: "
+            + PREFIX_NRIC + "NRIC "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_DOB + "DOB "
+            + PREFIX_PHONE + "PHONE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NRIC + "S1234567A "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_DOB + "2003-01-30 "
+            + PREFIX_PHONE + "98765432";
+
+    public static final String MESSAGE_SUCCESS = "New patient added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This patient already exists in the address book";
+
+    private final Patient toAdd;
+
+    /**
+     * Creates an AddCommand to add the specified {@code Person}
+     */
+    public AddPatientCommand(Patient patient) {
+        requireNonNull(patient);
+        toAdd = patient;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (model.hasPerson(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        model.addPerson(toAdd);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof AddCommand)) {
+            return false;
+        }
+
+        AddPatientCommand otherAddCommand = (AddPatientCommand) other;
+        return toAdd.equals(otherAddCommand.toAdd);
```

Comment:
Good equals method! Especially typecasting instead of assuming that its a AddPatientCommand